picoquic/sender.c

```diff
@@ -1230,7 +1267,51 @@ picoquic_packet_type_enum picoquic_packet_type_from_epoch(int epoch)
     return ptype;
 }
 
-/* Prepare the next packet to send when in one the client initial states */
+/* Prepare a required repetition or ack  in a previous context */
+uint32_t picoquic_prepare_packet_old_context(picoquic_cnx_t* cnx, picoquic_packet_context_enum pc,
+    picoquic_path_t * path_x, picoquic_packet* packet, uint64_t current_time, uint32_t * header_length)
+{
+    int is_cleartext_mode = (pc == picoquic_packet_context_initial) ? 1 : 0;
+    uint32_t length = 0;
+    size_t data_bytes = 0;
+    uint32_t checksum_overhead = picoquic_get_checksum_length(cnx, is_cleartext_mode);
+
+    *header_length = 0;
+
+    length = picoquic_retransmit_needed(cnx, pc, path_x, current_time, packet, 
+        &is_cleartext_mode, header_length);
+    
+    if (length == 0 && cnx->pkt_ctx[pc].ack_needed != 0) {
+        packet->ptype =
+            (pc == picoquic_packet_context_initial) ? picoquic_packet_initial :
+            ((pc == picoquic_packet_context_handshake) ? picoquic_packet_handshake :
```

Comment:
The extra () about (x?y:z) Seem unnecessary and actually detract from the idea that this is basically a switch statement.
This construct is very idiomatic and therefore clear:
```
x=a?x:
x=b?y:
    z;
```

Reply:
OK, if you like it better. But then you are relying on the parsing precedence in compilers, and I have been bitten by that before.

Reply:
Yah, then I type "C" into Google, it learned enough to autocomplete for me: "C operator precedence". :)   Nowadays, you'd expect all compilers to get these right.

Once thing I remember is to always quote binary bit operators, because it is just a bug in the spec -- their precedence is with the logical operators (&&, ||) instead of the more natural binary arithmetic operators!

Other than this, I never quote anything inside ?: -- its precedence is super-low.